test-case-reporting/src/test_result_record.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2020 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable @typescript-eslint/camelcase */
+import {DB, TestStatus, Travis} from 'test-case-reporting';
+import {Database} from './db';
+import md5 from 'md5';
+
+export class TestResultRecord {
+  constructor(private db: Database) {}
+
+  async insertTravisBuild(build: Travis.Build): Promise<number> {
+    const [buildId] = await this.db('builds')
+      .insert({
+        commit_sha: build.commitSha,
+        build_number: build.buildNumber,
+      } as DB.Build)
+      .returning('id');
+
+    return buildId;
+  }
+
+  async insertTravisJob(job: Travis.Job, buildId: number): Promise<number> {
+    const [jobId] = await this.db('jobs')
+      .insert({
+        build_id: buildId,
+        job_number: job.jobNumber,
+        test_suite_type: job.testSuiteType,
+      } as DB.Job)
+      .returning('id');
+
+    return jobId;
+  }
+
+  testStatus(skipped: boolean, success: boolean): TestStatus {
+    if (skipped) {
+      return 'SKIP';
+    }
+    return success ? 'PASS' : 'FAIL';
+  }
+
+  makeTestCaseName(description: string, suite: Array<string>): string {
```

Comment:
super-nit: since the suite bits come first in the output, I'm inclined to say lets make the suite be the first argument.

Reply:
Done.